packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -187,15 +189,24 @@ class _ProductQueryPageState extends State<ProductQueryPage>
                       padding: const EdgeInsetsDirectional.only(
                         start: SMALL_SPACE,
                       ),
-                      child: FloatingActionButton(
-                        backgroundColor: themeData.colorScheme.secondary,
-                        onPressed: () {
-                          _scrollToTop();
-                        },
-                        tooltip: appLocalizations.go_back_to_top,
-                        child: Icon(
-                          Icons.arrow_upward,
-                          color: themeData.colorScheme.onSecondary,
+                      child: SizedBox(
+                        height: MINIMUM_TOUCH_SIZE,
+                        child: ElevatedButton(
```

Comment:
> I guess `IconButton` would make more sense and the code would be less verbose. Would you please try it?

If we do use the `IconButton`, I believe it will need to be embedded inside an `InkResponse` widget for a circular ripple effect and inside a `Material` widget for background and elevation.

Perhaps something like this:

 ```
     ClipOval(
        child: Material(
          elevation: 2.0,
          color: themeData.colorScheme.secondary,
          child: InkResponse(
            containedInkWell: true,
            child: SizedBox(
              height: MINIMUM_TOUCH_SIZE,
              child: IconButton(
                icon: const Icon(
                  Icons.arrow_upward,
                ),
                color: themeData.colorScheme.onSecondary,
                onPressed: onPressed,
              ),
            ),
          ),
        ),
      );
```

Perhaps it could be added inside a private method that returns a widget:

```
Widget _getBackToTopButton(ThemeData themeData, void Function()? onPressed) =>
      ClipOval(
        child: Material(
          elevation: 2.0,
          color: themeData.colorScheme.secondary,
          child: InkResponse(
            containedInkWell: true,
            child: SizedBox(
              height: MINIMUM_TOUCH_SIZE,
              child: IconButton(
                icon: const Icon(
                  Icons.arrow_upward,
                ),
                color: themeData.colorScheme.onSecondary,
                onPressed: onPressed,
              ),
            ),
          ),
        ),
      );
```

What do you think?

Reply:
@gmmoraes My assumption was that it was easy to use (and to maintain) an `IconButton` in order to display a clickable disk with an icon inside. Easier than an `ElevatedButton`.
If it's not the case, don't waste time with the hard to read/maintain solution you've just suggested, and ignore my suggestion about using an `IconButton`.